### DISCUSSION TITLE
Microchip PolarFire SoC Yocto BSP - Linux Build

### DISCUSSION THREAD

**Discussion by Pronadeep2024** [polarfire-soc/polarfire-soc-documentation#398] [Linux build systems for PolarFire SoC] [answered]
In the demo PolarFire SoC FPGA H.264 Video Streaming Over Ethernet (AN4529) the Linux image - (core-image-minimal-devmpfs-video-kit-xxxx.rootfs.wic) is running on single RISC-V core or all the 4 RISC-V Cores of PolarFire SoC. Please provide more insight on the Linux build.

**Comment by sc-mchp** [comment, marked as answer]
Hi Pronadeep,

1)	(AN4529) is having old tag v2023.02 images Linux version 5.15, the latest image is there v2023.09 new Linux version 6.1.
2)	The H264 demo uses 32% of one CPU, further it is not bound to one CPU (keeps switching between available cores).
3)	The Demo uses FFMPEG open-source code to stream H.264 frames.
4)	FPGA fabric will do H.264 compression, then FFMPEG running in MSS packages it and sends it over the ethernet.

Thanks,
Shravan

**Reply by Pronadeep2024** [reply]
Hi Shravan, Thanks for the reply.

From, 2nd point that you mean to say Linux will be running on the RISC cores in SMP ( Symmetric Multiprocessing ) mode.

Any, documentation on the Linux image that I can refer.

Thank You

**Reply by sc-mchp** [reply]
This is more generic query to wared RISC-V,
I found some linux support for SMP
https://github.com/linux4microchip/linux/blob/linux-6.1-mchp/arch/riscv/kernel/smpboot.c
https://github.com/linux4microchip/linux/blob/linux-6.1-mchp/arch/riscv/kernel/smp.c

Thanks,

**Reply by Pronadeep2024** [reply]
Thanks for the direction :)
